### PR TITLE
Dockerfile: switched to distroless, specified defaults environment variables for containerized kopia

### PR DESCRIPTION
**Pull request by jkowalski**

### CODE REVIEW

File: tools/docker/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,23 @@
 FROM alpine
 ARG TARGETARCH
-RUN apk add --no-cache --verbose ca-certificates && adduser -D kopia && addgroup kopia kopia
+
+# install CA certificates and create user/group 'kopia' with IDs 51515:51515
+RUN apk add --no-cache --verbose ca-certificates && \
```
Comment (jeremyje) [inline tools/docker/Dockerfile:5]: What are the certs for?

File: tools/docker/Dockerfile [hunk 2]
```diff
@@ -1,6 +1,23 @@
 FROM alpine
```
Comment (jeremyje) [inline tools/docker/Dockerfile:1]: Consider distroless:static-nonroot if this is just a go binary

**Review by jeremyje — CHANGES_REQUESTED** [review]

File: tools/docker/Dockerfile [hunk 3]
```diff
@@ -1,6 +1,23 @@
...
 ARG TARGETARCH
-RUN apk add --no-cache --verbose ca-certificates && adduser -D kopia && addgroup kopia kopia
+
+# install CA certificates and create user/group 'kopia' with IDs 51515:51515
+RUN apk add --no-cache --verbose ca-certificates && \
+  addgroup -g 51515 kopia && \
+  adduser -u 51515 -H -D -G kopia kopia && \
+  mkdir -p /logs /cache /config && \
```
Comment (jeremyje) [inline tools/docker/Dockerfile:8]: Kopia should be doing the directory creations in place of it needs them.

Comment (jeremyje) [inline tools/docker/Dockerfile:8]: Kopia should be doing the directory creations in place of it needs them.

File: tools/docker/Dockerfile [hunk 4]
```diff
@@ -1,6 +1,23 @@
...
+  chown kopia:kopia /logs /cache /config
```
Comment (jeremyje) [inline tools/docker/Dockerfile:9]: Unless there's good reason I'd not create a bunch of directories in the root. 

The script here can be simplified (ideally no RUN statements). That enables miltiarch building

**Review by jeremyje — CHANGES_REQUESTED** [review]

File: tools/docker/Dockerfile (hunk 1 above)
Comment (jkowalski) [reply tools/docker/Dockerfile:5]: mostly connecting to S3, Azure and other services on the internet.

File: tools/docker/Dockerfile (hunk 2 above)
Comment (jkowalski) [reply tools/docker/Dockerfile:1]: it actually uses one Unix command to set TTY parameters, I thought about it, perhaps if we can remove this dependency, we can move to distroless.

File: tools/docker/Dockerfile (hunk 4 above)
Comment (jkowalski) [reply tools/docker/Dockerfile:9]: this is already multiarch for 3 linux archs, Windows won't work, though.

File: tools/docker/Dockerfile (hunk 3 above)
Comment (jkowalski) [reply tools/docker/Dockerfile:8]: unless those are under `/home/kopia` (which is ugly and I don't want to create it) it won't have permissions.

**Comment by jkowalski** [comment]
Switched to `gcr.io/distroless/static:nonroot`, PTAL @jeremyje

File: tools/docker/Dockerfile (hunk 2 above)
Comment (jkowalski) [reply tools/docker/Dockerfile]: fixed that.

File: tools/docker/Dockerfile (hunk 3 above)
Comment (jkowalski) [reply tools/docker/Dockerfile]: got rid of all `RUN`

**Review by jeremyje — APPROVED** [review]

**Pull request merged**